src/functional/fencil_processors/callables/cpp/bindings.py

```diff
@@ -0,0 +1,195 @@
+import os
+from typing import Any, Sequence, Type, TypeVar
+
+import jinja2
+
+import eve.codegen
+import functional.fencil_processors.cpp as cpp
+import functional.fencil_processors.defs as defs
+from eve import Node
+from eve.codegen import JinjaTemplate, TemplatedGenerator
+
+
+T = TypeVar("T")
+
+
+class CommaSeparatedList(Node):
+    items: list
+
+
+class NewlineSeparatedList(Node):
+    items: list
+
+
+class BindingFunction(Node):
+    exported_name: str
+    wrapper_name: str
+    doc: str
+
+
+class BindingModule(Node):
+    name: str
+    doc: str
+    functions: NewlineSeparatedList
+
+
+class FunctionParameter(Node):
+    name: str
+    ndim: int
+    dtype: Type
+
+
+class FunctionCall(Node):
+    target: defs.Function
```

Comment:
They have very different semantics. `defs.Function` defines the signature of a callable and is used exchange information between code generators and binding generators, `FunctionCall` is specific to the bindings and defines a call to a `defs.Function`. `FunctionCall` could also contain arbitrary arguments to the call, but in this case the arguments are not needed because they can be completely figured out from the signature. Using the same object for both would be confusing to me.

Reply:
Yes, I agree. I was confused by the lack of `args` field. By the way, shouldn't `args` be some kind of collection (`Sequence` or `Mapping`) of `Any`s?

Reply:
In the end I added the args field because of moving stuff into the node visitor. Yeah, it should be Sequence[Any], it's fixed now.